lib/double-fetch-detection.c

```diff
@@ -0,0 +1,34 @@
+#include "linux/double-fetch-detection.h"
+#include <linux/context_tracking.h>
+#include <linux/entry-common.h>
+#include <linux/livepatch.h>
+#include <linux/audit.h>
+#include <linux/sched.h>
+#include <linux/slab.h>
+void addAddress(void* addr, unsigned long leng){
+           if(current->noRead==100 || current->noRead==-1)
+             return;
+           ((current->addresses)+current->noRead)->startaddress = addr;
+            ((current->addresses)+current->noRead)->len =leng;
+           current->noRead++;
+}
+
+void startSystemCall(){
+    current->addresses= (struct address*)kmalloc_array(100,sizeof(struct address),GFP_KERNEL);
+    if(!current->addresses)
+        current->noRead=-1;
+    else
+        current->noRead=0;
+
+}
+void endSysCall(){
+    print();
```

Comment:
Shouldn't this warn if you reach some threshold of read accesses? Right now it seems it always prints something if noRead>0.

Will this still change?

Reply:
Print function i used only for testing or printing anything i need during debugging.